src/main/java/org/spongepowered/server/world/chunkio/AsyncAnvilChunkLoader.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.server.world.chunkio;
+
+import static net.minecraft.world.chunk.storage.AnvilChunkLoader.readChunkEntity;
+
+import net.minecraft.block.Block;
+import net.minecraft.nbt.CompressedStreamTools;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.nbt.NBTTagList;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.datafix.FixTypes;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.ChunkPos;
+import net.minecraft.world.World;
+import net.minecraft.world.chunk.Chunk;
+import net.minecraft.world.chunk.storage.AnvilChunkLoader;
+import net.minecraft.world.chunk.storage.RegionFileCache;
+
+import java.io.DataInputStream;
+import java.io.IOException;
+
+public final class AsyncAnvilChunkLoader {
+
+    private AsyncAnvilChunkLoader() {
+    }
+
+    public static Object[] load(AnvilChunkLoader loader, World worldIn, int x, int z) throws IOException {
```

Comment:
As long as it's documented, I'm fine with it, but generally, if we are adding implementation as dependent as this, I'd javadoc it, no matter whether it's borrowed code or not. Javadocs will save time from people in the future from having to test and understand the code fully to modify for future implementations (providing that future implementations are changing the methods ever so slightly.).

In general I'd keep as much documented, regardless of the origination of where the code came from, it'll help save time in the future for future updates between minecraft versions and minecraft variants within the supported versions.